src/handlers/auth.ts

```diff
@@ -24,28 +24,55 @@ import { HandlerError } from '../utils/errors';
  *     } catch (error) {
  *       // Add you own custom error logging.
  *       errorReporter(error);
- *       res.status(error.status || 500).end(error.message);
+ *       res.status(error.status || 500).end();
  *     }
  *   }
  * });
  * ```
  *
+ * Alternatively, you can customize the default handlers without overriding them. For example:
+ *
+ * ```js
+ * // pages/api/auth/[...auth0].js
+ * import { handleAuth, handleLogin } from '@auth0/nextjs-auth0';
+ *
+ * export default handleAuth({
+ *   login: handleLogin({
+ *     authorizationParams: { customParam: 'foo' } // Pass in custom params
+ *   })
+ * });
+ * ```
+ *
+ * You can also create new handlers by customizing the default ones. For example:
+ *
+ * ```js
+ * // pages/api/auth/[...auth0].js
+ * import { handleAuth, handleLogin } from '@auth0/nextjs-auth0';
+ *
+ * export default handleAuth({
+ *   signup: handleLogin({
+ *     authorizationParams: { screen_hint: 'signup' }
+ *   })
+ * });
+ * ```
+ *
  * @category Server
  */
-export interface Handlers {
-  login: HandleLogin;
-  logout: HandleLogout;
-  callback: HandleCallback;
-  profile: HandleProfile;
-  onError: OnError;
-}
+export type Handlers = ApiHandlers | ErrorHandlers;
+
+type ApiHandlers = {
+  [key: string]: NextApiHandler;
```

Comment:
Added types to this in [b02f9b4](https://github.com/auth0/nextjs-auth0/pull/840/commits/b02f9b4be2bf34a735c6a89920b16499eb0abb91)

Also noticed that the `login`, `logout` etc types weren't `HandleLogin`, `HandleLogout` but `NextApiHandler`.

<img width="847" alt="image" src="https://user-images.githubusercontent.com/1299658/194284329-c7433996-54c2-4c1b-8d33-2d171f564ccb.png">